### ISSUE TITLE
unable to spin up pihole:latest, pihole:5.1.2 with same compose works

### ISSUE THREAD

**Issue by hugalafutro** [pi-hole/docker-pi-hole#709]
Hi, my `docker-compose.yml` is:
```
services:
  pihole:
    container_name: pihole3
    environment:
      CONDITIONAL_FORWARDING: "true"
      CONDITIONAL_FORWARDING_DOMAIN: lan
      CONDITIONAL_FORWARDING_IP: 192.168.1.1
      CONDITIONAL_FORWARDING_REVERSE: 1.168.192.in-addr.arpa
      DNS1: 192.168.1.1
      DNS2: "no"
      ServerIP: 192.168.1.212
      TZ: Europe/London
      VIRTUAL_HOST: fr24
      WEBPASSWORD: <redacted>
    hostname: pihole3-docker
    image: pihole/pihole:latest
    ports:
    - 53:53/tcp
    - 53:53/udp
    - 67:67/udp
    - 7070:80/tcp
    restart: unless-stopped
    volumes:
    - /home/pi/docker/pi-hole/etc-pihole:/etc/pihole:rw
    - /home/pi/docker/pi-hole/etc-dnsmasq.d:/etc/dnsmasq.d:rw
version: '3.0'
```
the image got updated few hours ago so I wanted to test upgrade, however it fails with:
```
Creating network "pi-hole_default" with the default driver
Creating pihole3 ... done
Attaching to pihole3
pihole3   | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
pihole3   | [s6-init] ensuring user provided files have correct perms...exited 0.
pihole3   | [fix-attrs.d] applying ownership & permissions fixes...
pihole3   | [fix-attrs.d] 01-resolver-resolv: applying...
pihole3   | [fix-attrs.d] 01-resolver-resolv: exited 0.
pihole3   | [fix-attrs.d] done.
pihole3   | [cont-init.d] executing container initialization scripts...
pihole3   | [cont-init.d] 20-start.sh: executing...
pihole3   |  ::: Starting docker specific checks & setup for docker pihole/pihole
  [✓] Update local cache of available packages
pihole3   |   [i] Existing PHP installation detected : PHP version 7.3.19-1~deb10u1
pihole3   |
pihole3   |   [i] Installing configs from /etc/.pihole...
pihole3   |   [i] Existing dnsmasq.conf found... it is not a Pi-hole file, leaving alone!
  [✓] Copying 01-pihole.conf to /etc/dnsmasq.d/01-pihole.conf
pihole3   | chown: cannot access '': No such file or directory
pihole3   | chmod: cannot access '': No such file or directory
pihole3   | chown: cannot access '/etc/pihole/dhcp.leases': No such file or directory
pihole3   | ::: Pre existing WEBPASSWORD found
pihole3   | Using custom DNS servers: 192.168.1.1 & no
pihole3   | DNSMasq binding to default interface: eth0
pihole3   | Added ENV to php:
pihole3   |                     "PHP_ERROR_LOG" => "/var/log/lighttpd/error.log",
pihole3   |                     "ServerIP" => "192.168.1.212",
pihole3   |                     "VIRTUAL_HOST" => "fr24",
pihole3   | Using IPv4 and IPv6
pihole3   | ::: Preexisting ad list /etc/pihole/adlists.list detected ((exiting setup_blocklists early))
pihole3   | https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
pihole3   | https://mirror1.malwaredomains.com/files/justdomains
pihole3   |
pihole3   | dnsmasq: error at line 42 of /etc/dnsmasq.d/01-pihole.conf
pihole3   | ::: Testing pihole-FTL DNS: [cont-init.d] 20-start.sh: exited 1.
pihole3   | [cont-finish.d] executing container finish scripts...
pihole3   | [cont-finish.d] done.
pihole3   | [s6-finish] waiting for services.
pihole3   | [s6-finish] sending all processes the TERM signal.
pihole3   | [s6-finish] sending all processes the KILL signal and exiting.
```
changing tag to 5.1.2 and recreating the container spins it up no problem.

this is the offending file `cat etc-dnsmasq.d/01-pihole.conf`:
```
# Pi-hole: A black hole for Internet advertisements
# (c) 2017 Pi-hole, LLC (https://pi-hole.net)
# Network-wide ad blocking via your own hardware.
#
# Dnsmasq config for Pi-hole's FTLDNS
#
# This file is copyright under the latest version of the EUPL.
# Please see LICENSE file for your rights under this license.

###############################################################################
#      FILE AUTOMATICALLY POPULATED BY PI-HOLE INSTALL/UPDATE PROCEDURE.      #
# ANY CHANGES MADE TO THIS FILE AFTER INSTALL WILL BE LOST ON THE NEXT UPDATE #
#                                                                             #
#        IF YOU WISH TO CHANGE THE UPSTREAM SERVERS, CHANGE THEM IN:          #
#                      /etc/pihole/setupVars.conf                             #
#                                                                             #
#        ANY OTHER CHANGES SHOULD BE MADE IN A SEPARATE CONFIG FILE           #
#                    WITHIN /etc/dnsmasq.d/yourname.conf                      #
###############################################################################

addn-hosts=/etc/pihole/local.list
addn-hosts=/etc/pihole/custom.list


localise-queries


no-resolv



cache-size=10000

log-queries
log-facility=/var/log/pihole.log

local-ttl=2

log-async
server=192.168.1.1
interface=eth0
rev-server=1.168.192.in-addr.arpa,192.168.1.1
server=/lan/192.168.1.1
server=/use-application-dns.net/
```

Any help appreciated, thanks!

**Comment by dschaper** [comment]
Do the changes explained with https://github.com/pi-hole/docker-pi-hole/pull/707/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R113 for the notation of `CONDITIONAL_FORWARDING_REVERSE:` fix the issue? 

Change `1.168.192.in-addr.arpa` to `192.168.1.0/24`

**Comment by hugalafutro** [comment]
That fixed it, many thanks!

**Comment by dschaper** [comment]
I'm going to pin this issue and clear the name up a little bit. I think there will be more people looking for this info.

**Comment by zhorvath83** [comment]
Please update the CONDITIONAL_FORWARDING_REVERSE environment variable documentation at https://hub.docker.com/r/pihole/pihole later too. Thanks!

**Comment by dschaper** [comment]
That variable is supposed to be deleted by the script: https://github.com/pi-hole/pi-hole/pull/3761/files#diff-96ca2289baea839d41151c62c1945b01bfba1281a89fe4175eca7d36fe3ba1d3R238

@diginc is there something we broke that prevents `CONDITIONAL_FORWARDING_REVERSE` from being removed as https://github.com/pi-hole/pi-hole/pull/3761 intends?

**Comment by diginc** [comment]
I'll have to dig more later - but I know ProcessDNSSettings is called during startup by https://github.com/pi-hole/docker-pi-hole/blob/master/bash_functions.sh#L142

**Comment by PromoFaux** [comment]
All, this should now be resolved in `v5.2.1` (And I've just tested it on my own install - so my _should_ is on the confident side)

`CONDITIONAL_FORWARDING_REVERSE=168.192.in-addr.arpa` is correctly translated to `REV_SERVER_CIDR=192.168.0.0/16`

Please note that 5.2.1 now also support directly setting the `REV_SERVER*` variables. The `CONDITIONAL_FORWARDING*` ones will still work, but they are now deprecated
